bin/utils.ml

```diff
@@ -0,0 +1,6 @@
+let file_contents file = In_channel.(with_open_bin file In_channel.input_all)
+
+let pretty_print_link_status (link, (status_code, status_string)) =
+  Printf.printf "%s - %i %s\n" link status_code status_string
+
+let with_arg f x = (x, f x)
```

Comment:
Usually, stuff like this `with_arg` are left close to the place where they are used, or used as anonymous functions. It's easier to read their code each time that remembering what `with_arg` stands for (arguably, this is not a great name)